### PR TITLE
Implement "form" slider bar control

### PR DESCRIPTION
**Pull request by bdach**
Continuation of https://github.com/ppy/osu/issues/29643.

https://github.com/user-attachments/assets/3642f3cc-57bd-4e04-b9f2-69f569cc4bff

### CODE REVIEW

**Comment by peppy** [comment]
I've held back on visual improvements with these controls so far. They feel pretty damn good except for hover state, where I'd hope for intermediate animations (when mouse is down) and also handling the edge case where you are still controlling an element while not hover it directly, ie. I'd expect this operation to not lose "active" colouring:

https://github.com/user-attachments/assets/6115b32c-584f-4820-8446-49c34a200284

File: osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+    public partial class FormSliderBar<T> : CompositeDrawable, IHasCurrentValue<T>
+        where T : struct, INumber<T>, IMinMaxValue<T>
+    {
+        public Bindable<T> Current
+        {
+            get => current.Current;
+            set => current.Current = value;
+        }
+
+        private bool instantaneous;
```
Comment (peppy) [inline osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs:32]: I'd hope the default for this to be `true` (like we'd want it to be instantaneous for every usage except ones where that doesn't work). Thoughts?

.. although I'm remembering someone wanting defaults to always be `false` back some years..

File: osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+            {
+                instantaneous = value;
+                slider.TransferValueOnCommit = !instantaneous;
+            }
+        }
+
+        private readonly BindableNumberWithCurrent<T> current = new BindableNumberWithCurrent<T>();
+
+        public LocalisableString Caption { get; init; }
+        public LocalisableString HintText { get; init; }
```
Comment (peppy) [inline osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs:51]: Probably document these just for completion sake (makes them stand out better as stuff that you'd care about when constructing).

**Comment by bdach** [comment]
> I'd expect this operation to not lose "active" colouring:

Sure, I'll see what I can do.

> I'd hope for intermediate animations (when mouse is down)

Can you expound on this a bit? What is it that you'd be looking for, a smoother transition between hover and "active" state?

File: osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Graphics/UserInterfaceV2/FormSliderBar.cs]: Have changed to default to true (and fixed implementation too because it was broken).

**Review by peppy — APPROVED** [review]

**Pull request merged**